ps2/pavla ps2.R

```diff
@@ -0,0 +1,213 @@
+# education-num: continuous.
+# marital-status: Married-civ-spouse, Divorced, Never-married, Separated, Widowed, Married-spouse-absent, Married-AF-spouse.
+# occupation: Tech-support, Craft-repair, Other-service, Sales, Exec-managerial, Prof-specialty, Handlers-cleaners, Machine-op-inspct, Adm-clerical, Farming-fishing, Transport-moving, Priv-house-serv, Protective-serv, Armed-Forces.
+# relationship: Wife, Own-child, Husband, Not-in-family, Other-relative, Unmarried.
+# race: White, Asian-Pac-Islander, Amer-Indian-Eskimo, Other, Black.
+# sex: Female, Male.
+# capital-gain: continuous.
+# capital-loss: continuous.
+# hours-per-week: continuous.
+# native-country: United-States, Cambodia, England, Puerto-Rico, Canada, Germany, Outlying-US(Guam-USVI-etc), India, Japan, Greece, South, China, Cuba, Iran, Honduras, Philippines, Italy, Poland, Jamaica, Vietnam, Mexico, Portugal, Ireland, France, Dominican-Republic, Laos, Ecuador, Taiwan, Haiti, Columbia, Hungary, Guatemala, Nicaragua, Scotland, Thailand, Yugoslavia, El-Salvador, Trinadad&Tobago, Peru, Hong, Holand-Netherlands.
+# salary: >50K,<=50K
+
+# Further description is available in adult.names file
+
+# 1) read the data, make sure columns are read in the right format, i.e. character, double, etc.
+
+library(magrittr)
+library(tidyverse)
+library(purrr)
+library(foreign)
+library(dplyr)
+library(stringr)
+
+adult <- read.csv("adult.data", header=F)
+as.data.frame(adult)
+
+# 2) How many men and women are represented in the dataset? 
+
+print(unique(adult$V10))
+# There are some blank spaces
+
+female <- adult %>%
+  filter(V10 == " Female") %>%
+  nrow()
+
+male <- adult %>%
+  filter(V10 == " Male") %>%
+  nrow()
+
+cat("There are", female, "women and", male, "men.")
+
+# 3) What is the average age of women
+
+adult %>%
+  filter(V10 == " Female") %>%
+  summarise(mean(V1))
+
+# 4) What are the mean and standard deviation of age for those who earn more than 50K per year
+# and those who earn less than 50K per year
+
+rich <- adult %>%
+  filter(V15 == " >50K") %>%
+  summarise(mean(V1, na.rm = T), sd(V1, na.rm = T)) 
+
+poor <- adult %>%
+  filter(V15 == " <=50K") %>%
+  summarise(mean(V1, na.rm = T), sd(V1, na.rm = T))
+
+rich
+poor
+
+# 5) Is it true that people who earn more than 50K have at least high school education? (education - 
+#     Bachelors, Prof-school, Assoc-acdm, Assoc-voc, Masters or Doctorate feature)
+
+adult %>%
+  filter(V15 == " >50K") %>%
+  count(V4)
+
+# No, as there are people with other types of education too.
+
+# 6) Display age statistics for each rate and each gender. Find the maximum age of men of 
+# Amer-Indian-Eskimo race
+
+adult %>%
+  group_by(V9) %>%
+  summarise(mean(V1), sd(V1), min(V1), max(V1), median(V1))
+
+adult %>%
+  group_by(V10) %>%
+  summarise(mean(V1), sd(V1), min(V1), max(V1), median(V1))
+
+adult %>%
+  filter(V10 == " Male", V9 == " Amer-Indian-Eskimo") %>%
+  summarise(max(V1))
+
```

Comment:
This is correct, but...
it is so complicated to understand what those V's in the column names stand for. Next time, I will subtract points for that.